src/components/tables/TableThunkActionCreators.ts

```diff
@@ -0,0 +1,151 @@
+import { ITableState, ITablesState } from './TableReducers';
+import { convertUndefinedAndNullToEmptyString } from '../../utils/FalsyValuesUtils';
+import { TABLE_PREDICATE_DEFAULT_VALUE, TableSortingOrder, TableChildComponent, DEFAULT_TABLE_DATA } from './TableConstants';
+import * as _ from 'underscore';
+import { ITableOwnProps, ITableHeadingAttribute } from './Table';
+import { turnOnLoading, turnOffLoading } from '../loading/LoadingActions';
+import { getTableLoadingIds, getTableChildComponentId } from './TableUtils';
+import { changeLastUpdated } from '../lastUpdated/LastUpdatedActions';
+import { setIsInError, modifyState } from './TableActions';
+import { addActionsToActionBar } from '../actions/ActionBarActions';
+import { unselectAllRows } from './TableRowActions';
+import * as $ from 'jquery';
+
+export const dispatchPreTableStateModification = (tableOwnProps: ITableOwnProps, dispatch: any) => {
+  dispatch(unselectAllRows(tableOwnProps.id));
+  dispatch(
+    addActionsToActionBar(
+      getTableChildComponentId(tableOwnProps.id, TableChildComponent.ACTION_BAR),
+      [],
+    ),
+  );
+  dispatch(turnOnLoading(getTableLoadingIds(tableOwnProps.id), tableOwnProps.id));
+};
+
+export const dispatchPostTableStateModification = (tableOwnProps: ITableOwnProps, dispatch: any) => {
+  dispatch(turnOffLoading(getTableLoadingIds(tableOwnProps.id), tableOwnProps.id));
+  dispatch(changeLastUpdated(getTableChildComponentId(tableOwnProps.id, TableChildComponent.LAST_UPDATED)));
+};
+
+export const defaultTableStateModifyer = (
+  tableOwnProps: ITableOwnProps,
+  shouldResetPage: boolean,
+): ((tableState: ITableState) => ITableState) => {
+  return (tableState: ITableState) => {
+    const tableDataById = tableState.data.byId;
+
+    let totalPages: number;
+    let totalEntries: number;
+    let nextDisplayedIds = [...tableState.data.allIds];
+
+    // predicates default logic
+    if (!_.isEmpty(tableState.predicates)) {
+      _.pairs(tableState.predicates).forEach((keyValuePair: string[]) => {
+        const attributeName = keyValuePair[0];
+        const attributeValue = keyValuePair[1];
+
+        if (attributeValue !== TABLE_PREDICATE_DEFAULT_VALUE) {
+          nextDisplayedIds = nextDisplayedIds.filter((dataId: string) =>
+            tableDataById[dataId][attributeName] === attributeValue);
+        }
+      });
+    }
+
+    // filter default logic
+    if (tableState.filter) {
+      const filterDefault = (dataId: string): boolean => {
+        let shouldKeep = false;
+
+        tableOwnProps.headingAttributes.forEach((headingAttribute: ITableHeadingAttribute) => {
+          const { attributeName, attributeFormatter } = headingAttribute;
+          const attributeValue = tableDataById[dataId][attributeName];
+          const attributeValueToUse = attributeFormatter ? attributeFormatter(attributeValue) : attributeValue;
+          shouldKeep =
+            shouldKeep
```

Comment:
You should wrap the content of your loop with `if (!shouldKeep) {` so that it does not compute everything every time if `shouldKeep` is set to true on the first time. You could also change it for a `for (const headingAttribute in  tableOwnProps.headingAttributes) {` and use a `break` once we set `shouldKeep` to true.

Reply:
It's a good use case for `_.some` instead of forEach

Reply:
definitely `some` is the thing to use here, fyi @sroy3 `shouldKeep = true || thisIsNotEvaluatedOrComputedSinceJSisLazylyEvaluated` only compute `shouldKeep = true` , the `for (const ...` would have been a good solution too but `some` is cleaner.